packages/date/src/DateRange.js

```diff
@@ -159,7 +158,7 @@ const DateRange = ({
 
   const onFocusChange = async input => {
     if (!input) await setFieldTouched(name, true);
-    if (autoSync) await syncDates();
+    if (autoSync && !endValue) await syncDates();
```

Comment:
And then in order to run autoSync we can either be `autoSync && (!input || input === 'endDate)`

Furthermore, we only want to call `setFieldTouched` inside of `syncDates` if we are unfocusing completely. So we should pass the `input` to the `syncDates` and the conditionally check if the input exists before calling touched.

```js
if(!input) // we are unfocusing completely
   setFieldTouched(name,true)
```

Reply:
Agree with this idea, but if we are calling `syncDates()` behind the condition `(!input && autoSync)` then I don't think we need to pass in `input` and do a second check

Reply:
It somewhat works only with `datepicker` but if you type into the field itself it actually doesn't autofill the start date back for you. So I would consider this a fix to make autoSync only work if you start with the `startdate`. Anything outside of that case is going to cause more problems trying to force it to work

Reply:
I do know the reason we started looking at this fix for autoSync is that our users would get upset if we removed it - it's quite often that they want `start` and `end` dates the same, and don't want to have to pick it twice.  So I think we definitely want to keep it, even if it's a headache to deal with the edge-cases.

Reply:
Right, but the extreme edge-cases like `I want to pick and end date first and then auto sync the start date` is outside the boundaries of what really we can support unless we want to take over `react-dates` and support that or build our own custom date framework to start making sure the validation is working as intended.

Reply:
One way around that problem could be using a single `<Date />` component by default and adding an option/checkbox to incorporate a `<DateRange />`. That could eliminate the need to code around all these edge cases.

Reply:
So I tried 
`if (autoSync && (!input || input === 'endDate'))`
and it broke the autoSync from endDate to startDate (apparently that worked if you picked in the picker, not typing/pasting).  Basically same result as the code in this PR.

However, I tried:
`if (autoSync && (!startValue || !endValue)) await syncDates();`
and it allowed me to autoSync from endDate back to startDate, if I used the picker.  I figure the only time we want to autoSync is when one of the values is empty anyway, so it seems semantically pretty okay to me.

Just from poking around, it seems like `input` is always defined in the cases that were causing issues - even if the user opened the calendar with the icon, `input` was st ill set to `startDate`

Reply:
So without changing anything inside `syncDates()`, this snippet inside `onFocusChange()` seems to get rid of the infinite loop that was happening when the old `focusedInput` value became equal to current `input` value. Syncing works starting at either input, syncing works when typing into `startDate`. This could be modified so that typing into `endDate` will sync to `startDate` if it doesn't have a value, but again we're just going further into edge cases.

```
    if (!input && !autoSync) await setFieldTouched(name, true);
    if (autoSync && !(focusedInput !== input )) await syncDates();
```

Reply:
> *Edited focusedInput === input for clarity.
I haven't looked into where yet, but some place in the code it looks like focusedInput falls out of sync and that's why it can sometimes be equal to input. I think if we find where then we could eliminate this check.